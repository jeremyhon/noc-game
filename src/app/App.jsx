import React from 'react'
import { Provider, connect } from 'react-redux'
import { BrowserRouter, Switch, Route } from 'react-router-dom'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

import store from 'store'
import Navbar from 'nav'
import Intern from 'intern'
import NotFound from 'notfound'
import About from 'about'
import Company from 'company'
import * as engine from 'lib/engine'
import { populateInterns, populateCompanies } from 'actions'
import './App.css'

export class App extends React.Component {
  componentDidMount = () => {
    //start your engineeesssss~
    engine.start()
    this.props.populateInterns(3)
    this.props.populateCompanies(3)
  }

  render() {
    return (
      <BrowserRouter>
        <div className="app">
            <Navbar />
            <Switch>
              <Route exact path="/" component={Intern} />
              <Route path="/about" component={About} />
              <Route path="/companies" component={Company} />
              <Route component={NotFound} />
            </Switch>
        </div>
      </BrowserRouter>
    )
  }
}

const mapDispatchToProps = {
  populateInterns,
  populateCompanies,
}

const ConnectedApp = connect(undefined, mapDispatchToProps)(App)

class Framework extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <MuiThemeProvider>
          <ConnectedApp />
        </MuiThemeProvider>
      </Provider>
    )
  }
}

export default Framework
