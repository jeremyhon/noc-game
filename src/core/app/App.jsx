import React from 'react'
import PropTypes from 'prop-types'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import Navbar from '../nav'
import loop from './engine'
import './App.css'

class App extends React.Component {
  static propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node,
    ]).isRequired,
    location: PropTypes.shape({
      pathname: PropTypes.string,
    }).isRequired,
  }

  componentDidMount = () => {
    window.requestAnimationFrame(loop)
  }

  render() {
    return (
      <MuiThemeProvider>
        <div className="App">
          <Navbar currentPage={this.props.location.pathname} />
          {this.props.children}
        </div>
      </MuiThemeProvider>
    )
  }
}

export default App
