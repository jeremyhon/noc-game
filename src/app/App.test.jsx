import React from "react"
import { mount, shallow } from "enzyme"
import injectTapEventPlugin from 'react-tap-event-plugin'
import App from "./App"

it("matches the snapshot", () => {
  expect(shallow(<App />)).toMatchSnapshot()
})

it("renders without crashing", () => {
  injectTapEventPlugin()
  mount(<App />)
});
