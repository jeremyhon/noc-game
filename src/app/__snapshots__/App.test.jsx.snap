// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <BrowserRouter>
        <MuiThemeProvider>
            <div
                className="app"
            >
                <withRouter(Navbar) />
                <Switch>
                    <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                    />
                    <Route
                        component={[Function]}
                        path="/about"
                    />
                    <Route
                        component={[Function]}
                        path="/companies"
                    />
                    <Route
                        component={[Function]}
                    />
                </Switch>
            </div>
        </MuiThemeProvider>
    </BrowserRouter>
</Provider>,
  "nodes": Array [
    <Provider
      store={
            Object {
                  "dispatch": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                  Symbol(observable): [Function],
                }
      }
>
      <BrowserRouter>
            <MuiThemeProvider>
                  <div
                        className="app"
                  >
                        <withRouter(Navbar) />
                        <Switch>
                              <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/"
                              />
                              <Route
                                    component={[Function]}
                                    path="/about"
                              />
                              <Route
                                    component={[Function]}
                                    path="/companies"
                              />
                              <Route
                                    component={[Function]}
                              />
                        </Switch>
                  </div>
            </MuiThemeProvider>
      </BrowserRouter>
</Provider>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "componentDidMount": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
>
          <BrowserRouter>
                    <MuiThemeProvider>
                              <div
                                        className="app"
                              >
                                        <withRouter(Navbar) />
                                        <Switch>
                                                  <Route
                                                            component={[Function]}
                                                            exact={true}
                                                            path="/"
                                                  />
                                                  <Route
                                                            component={[Function]}
                                                            path="/about"
                                                  />
                                                  <Route
                                                            component={[Function]}
                                                            path="/companies"
                                                  />
                                                  <Route
                                                            component={[Function]}
                                                  />
                                        </Switch>
                              </div>
                    </MuiThemeProvider>
          </BrowserRouter>
</Provider>,
        "_debugID": 2,
        "_renderedOutput": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
>
          <BrowserRouter>
                    <MuiThemeProvider>
                              <div
                                        className="app"
                              >
                                        <withRouter(Navbar) />
                                        <Switch>
                                                  <Route
                                                            component={[Function]}
                                                            exact={true}
                                                            path="/"
                                                  />
                                                  <Route
                                                            component={[Function]}
                                                            path="/about"
                                                  />
                                                  <Route
                                                            component={[Function]}
                                                            path="/companies"
                                                  />
                                                  <Route
                                                            component={[Function]}
                                                  />
                                        </Switch>
                              </div>
                    </MuiThemeProvider>
          </BrowserRouter>
</Provider>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
